clients/drcachesim/reader/reader.cpp

```diff
@@ -429,7 +429,9 @@ reader_t::pre_skip_instructions()
         if (input_entry_->type != TRACE_TYPE_MARKER ||
             input_entry_->size == TRACE_MARKER_TYPE_TIMESTAMP) {
             // Likely some mock in a test with no page size header: just move on.
-            queue_.push(*input_entry_);
+            std::queue<trace_entry_t> entry;
+            entry.push(*input_entry_);
+            queue_to_return_next(entry);
```

Comment:
Not a fan of having to create a queue for one entry: make an overload of queue_to_return_next that takes a single entry?